sdk/iOS/test/MSSyncTableTests.m

```diff
@@ -1821,6 +1821,47 @@ -(void) testIncrementalPullWithInsert
     XCTAssertEqual(client.syncContext.pendingOperationsCount, 1);
 }
 
+// Tests incremental pull with interleaved local operations is Id case insensitive
+-(void)testIncrementalPullIsIdCaseInsensitive
+{
+    // Response of a pull operation
+    NSString *firstPullData = @"[{\"id\": \"one\", \"text\":\"first item\", \"__updatedAt\":\"1999-12-03T15:44:29.0Z\"},{\"id\": \"two\", \"text\":\"second item\", \"__updatedAt\":\"1999-12-03T15:44:28.0Z\"}]";
+
+    // Setup filter to get the pull response
+    MSMultiRequestTestFilter *filter = [MSMultiRequestTestFilter testFilterWithStatusCodes:@[@200] data:@[firstPullData] appendEmptyRequest:YES];
+
+    // Initialization
+    MSClient *filteredClient = [client clientWithFilter:filter];
+    MSSyncTable *todoTable = [filteredClient syncTableWithName:@"TodoItem"];
+    MSQuery *query = [[MSQuery alloc] initWithSyncTable:todoTable];
+
+    // Perform an insert operation in the middle of a pull operation
+    ((MSTestFilter *)filter.testFilters[0]).onInspectRequest = ^(NSURLRequest *request) {
+
+        // Insert an item with Id differing from that of the pulled item only in case
+        NSDictionary *item = @{ @"id": @"ONE", @"name":@"fifth item" };
+        MSTestWaiter *insertWaiter = [MSTestWaiter new];
```

Comment:
I was not aware of XCTestExpectation! :+1:  I actually implemented something similar and was planning to send a PR from another branch.
XCTestExpectation does not seem to support nested fulfills/waits. Here we want to wait for the pull operation to complete and independently wait for the insert operation to complete. Do you know if there is a way to achieve this?